src/basic/unit-name.c

```diff
@@ -797,3 +797,17 @@ bool slice_name_is_valid(const char *name) {
 
         return true;
 }
+
+bool unit_name_prefix_equal(const char *a, const char *b) {
+        const char *p, *q;
+
+        p = strchr(a, '@');
```

Comment:
strchr bombs out if the string is null - hence either assert or check and return false if one/both are null

Reply:
Added asserts